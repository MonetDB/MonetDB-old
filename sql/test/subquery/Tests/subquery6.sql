CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
CREATE TABLE integers(i INTEGER);
INSERT INTO integers VALUES (1), (2), (3), (NULL);

DELETE FROM another_t WHERE (SELECT 1 UNION SELECT 2) > 1; 
	--error, more than one row returned by a subquery used as an expression

DELETE FROM another_t WHERE (SELECT 1 UNION SELECT 2) > 1; 
	--error, more than one row returned by a subquery used as an expression

UPDATE another_T SET col1 = 1 WHERE (SELECT 1 UNION SELECT 2) > 1;
	--error, more than one row returned by a subquery used as an expression

MERGE INTO another_t USING (SELECT col1 FROM another_t) sub ON (SELECT 1 UNION SELECT 2) > 1 WHEN MATCHED THEN DELETE WHEN NOT MATCHED THEN INSERT;
	--error, more than one row returned by a subquery used as an expression

MERGE INTO another_t USING (SELECT (SELECT 1 UNION SELECT 2) FROM another_t) sub ON TRUE WHEN MATCHED THEN DELETE WHEN NOT MATCHED THEN INSERT;
	--error, more than one row returned by a subquery used as an expression

DROP TABLE tbl_ProductSales;
DROP TABLE another_T;
DROP TABLE integers;
