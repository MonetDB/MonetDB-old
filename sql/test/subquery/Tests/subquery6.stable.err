stderr of test 'subquery6` in directory 'sql/test/subquery` itself:


# 11:45:43 >  
# 11:45:43 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-297142" "--port=31114"
# 11:45:43 >  

MAPI  = (monetdb) /var/tmp/mtest-297142/.s.monetdb.31114
QUERY = DELETE FROM another_t WHERE (SELECT 1 UNION SELECT 2) > 1; 
ERROR = !GDK reported error: BATsubcross: more than one match
MAPI  = (monetdb) /var/tmp/mtest-297142/.s.monetdb.31114
QUERY = DELETE FROM another_t WHERE (SELECT 1 UNION SELECT 2) > 1; 
ERROR = !GDK reported error: BATsubcross: more than one match
MAPI  = (monetdb) /var/tmp/mtest-307385/.s.monetdb.36125
QUERY = UPDATE another_T SET col1 = 1 WHERE (SELECT 1 UNION SELECT 2) > 1;
ERROR = !GDK reported error: BATsubcross: more than one match
MAPI  = (monetdb) /var/tmp/mtest-307385/.s.monetdb.36125
QUERY = MERGE INTO another_t USING (SELECT col1 FROM another_t) sub ON (SELECT 1 UNION SELECT 2) > 1 WHEN MATCHED THEN DELETE WHEN NOT MATCHED THEN INSERT;
ERROR = !GDK reported error: BATsubcross: more than one match
MAPI  = (monetdb) /var/tmp/mtest-307385/.s.monetdb.36125
QUERY = MERGE INTO another_t USING (SELECT (SELECT 1 UNION SELECT 2) FROM another_t) sub ON TRUE WHEN MATCHED THEN DELETE WHEN NOT MATCHED THEN INSERT;
ERROR = !GDK reported error: BATsubcross: more than one match

# 11:45:43 >  
# 11:45:43 >  "Done."
# 11:45:43 >  

