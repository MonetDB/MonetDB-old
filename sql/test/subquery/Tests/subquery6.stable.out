stdout of test 'subquery6` in directory 'sql/test/subquery` itself:


# 11:45:43 >  
# 11:45:43 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-297142" "--port=31114"
# 11:45:43 >  

#CREATE TABLE tbl_ProductSales (ColID int, Product_Category  varchar(64), Product_Name  varchar(64), TotalSales int); 
#INSERT INTO tbl_ProductSales VALUES (1,'Game','Mobo Game',200),(2,'Game','PKO Game',400),(3,'Fashion','Shirt',500),(4,'Fashion','Shorts',100);
[ 4	]
#CREATE TABLE another_T (col1 INT, col2 INT, col3 INT, col4 INT, col5 INT, col6 INT, col7 INT, col8 INT);
#INSERT INTO another_T VALUES (1,2,3,4,5,6,7,8), (11,22,33,44,55,66,77,88), (111,222,333,444,555,666,777,888), (1111,2222,3333,4444,5555,6666,7777,8888);
[ 4	]
#CREATE TABLE integers(i INTEGER);
#INSERT INTO integers VALUES (1), (2), (3), (NULL);
[ 4	]
#DROP TABLE tbl_ProductSales;
#DROP TABLE another_T;
#DROP TABLE integers;

# 11:45:43 >  
# 11:45:43 >  "Done."
# 11:45:43 >  

